twitter_pr.rs

```diff
@@ -0,0 +1,96 @@
+extern crate reqwest;
+
+use std::sync::mpsc::{Receiver, channel};
+use serde::{Serialize, Deserialize};
+use tokio::stream::StreamExt;
+use tokio::io::{AsyncBufReadExt, StreamReader};
+use serde_json::{Deserializer, StreamDeserializer};
+use std::io::Write;
+
+pub struct TwitterApi {
+    pub bearer_api_token: String
+}
+
+struct RequestStream {
+    receiver: Receiver<bytes::Bytes>
+}
+
+impl std::io::Read for RequestStream {
+    fn read(&mut self, mut buf: &mut [u8]) -> std::io::Result<usize> {
+        match self.receiver.recv() {
```

Comment:
```suggestion
        match self.receiver.recv() {
```

Since this function returns a `Result`, you can (in theory) simplify this match with:

```
let next = self.receiver.recv()?;
```

I.e. use the [? operator](https://doc.rust-lang.org/edition-guide/rust-2018/error-handling-and-panics/the-question-mark-operator-for-easier-error-handling.html).

However, there's a bit of a hitch: This function returns a `Result<..., io::Error>`, and `recv()` returns a `Result<..., RecvError>`. Rust can automatically turn one into the other so long as there's an implementation of `Into<io::Error> for RecvError`. It looks like there isn't one, and the rules about defining traits on other packages' types means you can't define one.

However, what you *can* do is define your own local error type, and implement `Into<MyError> for RecvError` and `Into<io::Error> for MyError`. However, that's pretty complicated!

There's another tweak you can apply here: since you are okay (for now) with panicking if there's an error, you can write:

```
let next = self.receiver.recv().expect("error receiving from channel");
```